src/main/java/training/endava/skeleton/Main.java

```diff
@@ -51,17 +51,140 @@ public static void main(String argv[]) {
+//                        .stream()
+//                        .map(y -> y.getName())
+//                        .reduce("", (a, b) -> a + b))
+//                .reduce("", (a, b) -> a + b);
+//        System.out.println(new Person(name, "email", "tel", 123456, companies.get(3)));
+
+        //ex6
+//        companies.stream().
+//                forEach(x -> x.getEmployeeList()
+//                        .stream()
+//                        .map(y -> y.getName())
+//                .forEach(System.out::println));
+        //ex7
+//        companies.stream()
+//                .forEach(x -> x.getEmployeeList()
+//                        .stream()
+//                        .map(y -> y.getPhoneNumber())
+//                        .forEach(System.out::println));
+//        System.out.println("distinct ----------------------------");
+//        companies.stream()
+//                .forEach(x -> x.getEmployeeList()
+//                        .stream()
+//                        .map(y -> y.getPhoneNumber())
+//                        .distinct()
+//                        .forEach(System.out::println));
+        //ex8
+//        Collector<Person, StringJoiner, String> collector =
+//                Collector.of(
+//                        () -> new StringJoiner("//", "Toate: ", " persoanele"),
+//                        (supp, person) -> supp.add(person.getName()),
+//                        StringJoiner::merge,
+//                        StringJoiner::toString
+//                );
+//        Optional<String> name = companies.stream()
+//                .map(x -> x.getEmployeeList()
+//                        .stream()
+//                        .collect(collector))
+//                .findFirst();
+//        System.out.println(name.get());
+
+        //ex9
+//        Optional<Integer> upperBound = Optional.of(new Integer(1000));
+//        ArrayList<Company> companies2 = (ArrayList<Company>) makeCompanies(upperBound);
+//        companies2 = (ArrayList<Company>) populateCompanies(companies2);
+//
+//        Long streamStart = System.nanoTime();
+//
+//        companies2.stream()
+//                .forEach(x -> x.getEmployeeList()
+//                        .stream()
+//                        .map(y -> y.getPhoneNumber())
+//                        .reduce("",(a,b)->a+b));
+//
+//        System.out.println("Stream took ms:" + (System.nanoTime() - streamStart)/10000000);
+//        Long ParallelStreamStart = System.nanoTime();
+//
+//        companies2.parallelStream()
+//                .forEach(x -> x.getEmployeeList()
+//                        .stream()
+//                        .map(y -> y.getPhoneNumber())
+//                        .reduce("",(a,b)->a+b));
+//        System.out.println("ParallelStream took ms:" + (System.nanoTime() - streamStart)/10000000);
+
     }
 
-    public static ArrayList<Person> makeList() {
+    public static List<Company> makeCompanies( Optional<Integer> bound) {
```

Comment:
no need to declare a parameter optional, you could for example overload this method with one that has no parameter